core/src/components/pagination/pagination.ts

```diff
@@ -399,6 +435,36 @@ export function createPagination(config?: PropsConfig<PaginationProps>): Paginat
 		return pages$().map((page) => ariaPageLabel(page, pageCount));
 	});
 
+	const pagesHrefs$ = computed(() => {
+		const pageLinkFactory = pageLink$();
+		const pageCount = pageCount$();
+		return Array.from({length: pageCount}, (_, index) => pageLinkFactory(index + 1));
+	});
+
+	const directionsHrefs$ = computed(() => {
+		const pagesHrefs = pagesHrefs$();
+		const pageIndex = page$() - 1;
+		return {
+			previous: pagesHrefs.at(pageIndex > 0 ? pageIndex - 1 : 0)!,
+			next: pagesHrefs.at(pageIndex + 1) ?? pagesHrefs.at(-1)!,
+		};
+	});
+
+	/*
+	 * Stop event propagation when href is the default value;
+	 * Update page number when navigation is in the same tab and stop the event propagation;
+	 * For navigations outside current browser tab let the default behavior, without updating the page number;
+	 */
+	function handleNavigation(pageNumber: number, event?: PaginationClickEvent) {
+		if (pagesHrefs$()[pageNumber - 1] === PAGE_LINK_DEFAULT) {
+			event?.preventDefault();
+		}
+		if (!event || !(event.ctrlKey || event.metaKey)) {
+			event?.preventDefault();
+			patch({page: pageNumber});
```

Comment:
You can rename `pageNumber` to `page` and just do `patch({page});`. (I know, it's micro optimization, but it's for free ;) )